### ISSUE TITLE
Explore class weighting

### ISSUE THREAD

**Issue by dbuscombe-usgs** [Doodleverse/segmentation_gym#72]
problem: model metrics and output are not ideal when there is a minority class.
0. mean IOU is not effective score when a class is missing, or large class imbalance
1. problem manifests in two situations. first, when a class is rare among samples, and second when a class is rare in a given scene

Proposal: 
0. config file includes variable 'class_weight_scene':true and  'class_weight_global':{} dict of class:frequency pairs
1. if 'class_weight_scene' is true, compute class weighting from label and apply weighting
2. if 'class_weight_global' is present in the config file, parse and apply weighting
3. see, for example, https://www.tensorflow.org/tutorials/images/segmentation#optional_imbalanced_classes_and_class_weights

**Comment by ebgoldstein** [comment]
see here: https://github.com/Doodleverse/doodleverse_utils/issues/7#issuecomment-1263021118

this can be done w/ loss weighting instead of/ in addition

**Comment by dbuscombe-usgs** [comment]
Latest doodleverse-utils https://pypi.org/project/doodleverse-utils/0.0.10/ combined with this PR https://github.com/Doodleverse/segmentation_gym/pull/90 incorporates loss weighting for Dice (only). New config file argument  "LOSS_WEIGHTS":

If false, no loss weighting
If true, losses are weighted based on inverse per-class frequency distribution
If a list of numbers, losses are weighted using that vector. It is assumed that the weights for each class are listed in order

**Comment by dbuscombe-usgs** [comment]
Note to add to the wiki

**Comment by ebgoldstein** [comment]
I think once the documentation is updated, we can closs this issue.. Loss weighting will likely solve the same problem as 'class weighting', and was easier to implement (class weighting would require rewriting make dataset, so is therefore less ideal (i.e., new dataset created every time, vs. loss weighting can use the same created dataset... )

**Comment by dbuscombe-usgs** [comment]
Agree. In my mind, loss weighting and class weighting were synonyms when we opened this issue. Prefer the term 'loss weighting'. I added a script in utilities in the last PR that could be modified by the user to do 'class weighting' for binary problems. Further workflows could be future examples (tweetorials, wiki page, etc)
